pkg/kubeconfig/kubeconfig.go

```diff
@@ -17,22 +17,124 @@ limitations under the License.
 package kubeconfig
 
 import (
+	"bytes"
 	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
 	"io/fs"
 	"net"
 	"os"
+	"time"
 
 	kubeoneapi "k8c.io/kubeone/pkg/apis/kubeone"
+	"k8c.io/kubeone/pkg/certificate"
 	"k8c.io/kubeone/pkg/executor"
 	"k8c.io/kubeone/pkg/executor/executorfs"
 	"k8c.io/kubeone/pkg/fail"
 	"k8c.io/kubeone/pkg/state"
 
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
+	clientapi "k8s.io/client-go/tools/clientcmd/api"
+	clientapipublic "k8s.io/client-go/tools/clientcmd/api/latest"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	pkiCAcrt = "/etc/kubernetes/pki/ca.crt"
+	pkiCAkey = "/etc/kubernetes/pki/ca.key"
+)
+
+func GenerateSuperAdmin(st *state.State) ([]byte, error) {
+	host, err := st.Cluster.Leader()
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := st.Executor.Open(host)
+	if err != nil {
+		return nil, err
+	}
+
+	caCertsPEM, err := fs.ReadFile(executorfs.New(conn), pkiCAcrt)
+	if err != nil {
+		return nil, err
+	}
+
+	caCerts, err := certutil.ParseCertsPEM(caCertsPEM)
+	if err != nil {
+		return nil, err
+	}
+	if len(caCerts) == 0 {
+		return nil, fmt.Errorf("no certificates found in %s", pkiCAcrt)
+	}
+
+	caKeyPEM, err := fs.ReadFile(executorfs.New(conn), pkiCAkey)
+	if err != nil {
+		return nil, err
+	}
+
+	possibleCAKey, err := keyutil.ParsePrivateKeyPEM(caKeyPEM)
+	if err != nil {
+		return nil, fail.Runtime(err, "parsing private key %s PEM", pkiCAkey)
+	}
+
+	caRSAKey, ok := possibleCAKey.(*rsa.PrivateKey)
+	if !ok {
+		return nil, fail.NewRuntimeError("type asserting rsa.PrivateKey type", "private key is not a RSA private key")
+	}
+
+	superAdminUserKey, err := certificate.NewPrivateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	certCfg := certutil.Config{
+		CommonName:   "short-lived-super-admin",
```

Comment:
I think we should make this customizable. AFAIK, this name is used for different purposes including the audit logging, so it might be important to our users. We can default to `short-lived-super-admin`, but there should be a flag where you can set this.